kf_lib_data_ingest/validation/data_validator.py

```diff
@@ -0,0 +1,435 @@
+"""
```

Comment:
I don't know where to put this so I'm putting it here. It seems that if you do not read the data in with an encoding like this: `read_df(f, encoding="utf-8-sig")`, then the validation results show failures when there shouldn't be. 

For example, I tried running the data validator on the `tests/data/DATASET1` directory with just `read_df(f)` and I see errors that should not be there: 
```
'errors': deque([{'from': ('FAMILY|ID', 'F12'),
                   'locations': {('FAMILY|ID', 'F12'): ['/Users/singhn4/Projects/pull-request-reviews/kf-lib-data-ingest/tests/data/validation/DATASET1/pf.csv']},
                   'to': []},
                  {'from': ('FAMILY|ID', 'F10'),
                   'locations': {('FAMILY|ID', 'F10'): ['/Users/singhn4/Projects/pull-request-reviews/kf-lib-data-ingest/tests/data/validation/DATASET1/pf.csv']},
                   'to': []},
                  {'from': ('FAMILY|ID', 'F13'),
                   'locations': {('FAMILY|ID', 'F13'): ['/Users/singhn4/Projects/pull-request-reviews/kf-lib-data-ingest/tests/data/validation/DATASET1/pf.csv']},
                   'to': []}]),
```

But if I run with the encoding set then I get the expected errors: 

```
'errors': deque([{'from': ('FAMILY|ID', 'F12'),
                   'locations': {('FAMILY|ID', 'F12'): ['/Users/singhn4/Projects/pull-request-reviews/kf-lib-data-ingest/tests/data/validation/DATASET1/pf.csv']},
                   'to': []}]),
```

If I print out the df after reading it in both ways (with/without encoding), it looks the same. So I guess somewhere else in the validation code something weird is happening as a result of the encoding. Seems like its in the graph building piece because the graph is missing edges where it should have them.

Reply:
I think there's a problem with the files.

Reply:
See https://github.com/kids-first/kf-lib-data-ingest/pull/493/commits/9e2ac72b916580681a6749b7d6342b18b13e9428

The diff looks like nothing changed, but if you do the same in a terminal you'd see e.g. 

```diff
--- a/tests/data/validation/DATASET1/pf.csv
+++ b/tests/data/validation/DATASET1/pf.csv
@@ -1,5 +1,5 @@
-<U+FEFF>PARTICIPANT|ID,FAMILY|ID
-P10,F10
-P11,
-,F12
+PARTICIPANT|ID,FAMILY|ID
+P10,F10
+P11,
```

Reply:
Probably read_df should just always use encoding="utf-8-sig" by default.

Reply:
See https://github.com/kids-first/kf-lib-data-ingest/pull/509